torch/ao/sparsity/experimental/data_sparsifier/base_data_sparsifier.py

```diff
@@ -0,0 +1,246 @@
+        data_groups = copy.deepcopy(state_dict['data_groups'])
+        container_state_dict = state_dict['container_state_dict']
+
+        if strict:
+            # if strict load -> then reset container
+            self._container = Container()
+
+        for name, state in states.items():
+            config_name = data_groups.get(name, None)
+            if config_name is None:
+                raise RuntimeError(f"Error loading {name}")
+
+            parametrized_name = f'parametrizations.{name}.original'
+          
+            parametrized = False
+            data = container_state_dict.get(name, None)
+            if name in container_state_dict:
+                # the parametrization was probably removed for this
+                data = container_state_dict.get(name)
+
+            elif parametrized_name in container_state_dict:
+                # so the weight was parametrized
+                data = container_state_dict.get(parametrized_name)
+                parametrized = True
+
+            if data is None:
+                raise RuntimeError(f"Error loading {name}")
+
+            param = nn.Parameter(data, requires_grad=False)
+            setattr(self._container, name, param)
+
+            if parametrized:
+                # register parameter if parametrized
+                mask = state.get('mask', torch.ones_like(data))
+                param_class = data_groups.get('parametrization', FakeSparsity)
+                parametrize.register_parametrization(self._container, name, param_class(mask))
+
+        if not strict:
+            states.update(self.state)
+            data_groups.update(self.data_groups)
+          
+        self.__setstate__({'state': states, 'data_groups': data_groups,
+                        'container_state_dict': self._container.state_dict()})
+
+    def squash_mask(self, *args, **kwargs):
+        r"""Squashes the sparse masks into the appropriate tensors.
+        """
+        for name in self.data_groups.keys():
+            parametrize.remove_parametrizations(self._container, name, leave_parametrized=True)
+       
+    def convert(self):
+        # TODO: Call the torch.ao.utils.convert in here
+        raise NotImplementedError('`convert` is not implemented. Please, use '
+                                  '`torch.ao.utils.convert` instead.')
+
+    def step(self, use_path=True):
+        if not self.enable_mask_update:
+            return
+        with torch.no_grad():
+            for name, config in self.data_groups.items():
+                # get non-sparsified data
+                data = self.get_data(name, return_sparsified=False)
+                # need name for the mask otherwise can directly pass mask?
+                self.update_mask(name, data, **config)
+
+    @abc.abstractmethod
+    def update_mask(self, name, data, **kwargs):
+        pass
+
+
+__all__ = [Dict, Any, FakeSparsity, List, Optional, Tuple, defaultdict, 'BaseDataSparsifier', 'Container']
```

Comment:
`__all__` should be inside the `torch/ao/sparsity/experimental/data_sparsifier/__init__.py`